packages/website/src/components/ErrorsViewer.tsx

```diff
@@ -0,0 +1,115 @@
+import React, { useCallback, useEffect, useState } from 'react';
+import type Monaco from 'monaco-editor';
+import type { ErrorItem } from './types';
+
+import styles from './ErrorsViewer.module.css';
+
+export interface ErrorsViewerProps {
+  readonly value?: ErrorItem[];
+}
+
+export interface ErrorBlockProps {
+  readonly item: ErrorItem;
+  readonly onClick: () => void;
+  readonly isLocked: boolean;
+}
+
+function severityClass(severity: Monaco.MarkerSeverity): string {
+  switch (severity) {
+    case 8:
+      return 'danger';
+    case 4:
+      return 'caution';
+    case 2:
+      return 'note';
+  }
+  return 'info';
+}
+
+function ErrorBlock({ item, onClick, isLocked }: ErrorBlockProps): JSX.Element {
+  return (
+    <div className={`admonition alert alert--${severityClass(item.severity)}`}>
+      <div className="admonition-content">
+        <div className="row row--no-gutters">
+          <div className="col col--12">
+            {item.message} {item.location}
+          </div>
+          {item.hasFixers && (
+            <div className="col col--12">
+              {item.fixers.map((fixer, index) => (
+                <div key={index} className={styles.fixer}>
+                  <span>&gt; {fixer.message}</span>
+                  <button
+                    className="button button--primary button--sm"
+                    disabled={isLocked}
+                    onClick={(): void => {
+                      fixer.fix();
+                      onClick();
+                    }}
+                  >
+                    fix
+                  </button>
+                </div>
+              ))}
+            </div>
+          )}
+        </div>
+      </div>
+    </div>
+  );
+}
+
+export default function ErrorsViewer({
+  value,
+}: ErrorsViewerProps): JSX.Element {
+  const [model, setModel] = useState<[string, ErrorItem[]][]>();
+
+  const [isLocked, setIsLocked] = useState<boolean>(false);
+
+  const onClick = useCallback(() => {
+    setIsLocked(true);
+  }, [isLocked]);
+
+  useEffect(() => {
+    if (value) {
+      setModel(
+        Object.entries(
+          value.reduce<Record<string, ErrorItem[]>>((acc, obj) => {
+            if (!acc[obj.group]) {
+              acc[obj.group] = [];
+            }
+            acc[obj.group].push(obj);
+            return acc;
+          }, {}),
+        ).sort(([a], [b]) => a.localeCompare(b)),
+      );
+    } else {
+      setModel(undefined);
+    }
+    setIsLocked(false);
+  }, [value]);
```

Comment:
isn't that stopping recalculation of this model when `isLocked` is changed?

Reply:
It won't stop recalculation, no. The component function will run every time a prop -here, i.e `value`- changes or a hook causes it to.

Reply:
no I get that, if reference of value is changing callback of `useEffect` is triggered, but it's not executed when `isLocked` is changing. thats the reason why we need to pass "dependencies"

Reply:
@armano2 Why is `setIsLocked` called within an effect specifically? Why do we want to delay the update to the next render? Is the effect intended to give time for monaco to update its model and reported errors?

Reply:
i did it to prevent user to apply multiple times the same fix, after fix is applied we have to wait for eslint and monaco to report us new errors

Reply:
Is there any way for us to thread some state down via monaco to track when moncao is ready?
I'm assuming not - but that would be preferable.

---

I would personally hoist the model outside of the effect so that it's not state.
```ts
const model = useMemo(
  () => value ? Object.entries(...).sort(...) : undefined,
  [value]
);
useEffect(() => { setIsLocked(false) }, [value]);
```

This just makes sure that the model value is always up-to-date based on the props whilst also delaying the unlock slightly to ensure that there's no way we'll accidentally queue two separate updates.